### PR TITLE
add support for pointerover, pointerout events

### PR DESCRIPTION
**Pull request by vbud**
It would be awesome if mjolnir supported more event types, so here is a PR adding support for `pointerenter`.

I'm a bit confused about how to enable `pointerenter` and `pointerleave` events though, it seems like they are disabled by default? https://github.com/uber-web/mjolnir.js/blob/master/src/event-manager.js#L115.

### CODE REVIEW

**Comment by vbud** [comment]
@Pessimistress any feedback on this PR?

I'm trying to determine if we can get `pointerenter` and `pointerleave` events via react-map-gl's `eventManager` via Mjolnir, or if we need to pursue wrapping the `eventManager` or creating a custom map control.

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "webpack-dev-server": "^2.4.0"
   },
   "engines": {
-    "node": ">= 4",
+    "node": ">= 4 <= 8",
```
Comment (Pessimistress) [inline package.json]: What's wrong with higher versions?

**Comment by Pessimistress** [comment]
> I'm a bit confused about how to enable `pointerenter` and `pointerleave` events though, it seems like they are disabled by default?

An event is only enabled when a handler is registered for it. It doesn't really matter for this one but has a perf implication for gesture events.

**Comment by vbud** [comment]
> An event is only enabled when a handler is registered for it. It doesn't really matter for this one but has a perf implication for gesture events.

That makes sense, thank you.

File: package.json (hunk 1 above)
Comment (vbud) [reply package.json]: It looks like `fsevents` 1.1.2 does not support node 9 and newer, but 1.1.3 does. I'll open a separate PR getting the appropriate packages upgraded.

Comment (vbud) [reply package.json]: Ok I have this working in https://github.com/uber-web/mjolnir.js/pull/27, I'll remove this change from this PR.

**Comment by vbud** [comment]
Testing locally it looks like we don't actually get `mouseenter` event types in the MoveInput handler. We also don't reliably get `mouseleave` events.

I'll see if I can work around this and update you when the PR is ready for review again.

**Comment by vbud** [comment]
@Pessimistress ok, this is ready for review. Changes since you last looked:
- I realized that what I want is `pointerover` and `pointerout` support, not `pointerenter`. So I updated the PR to add support for those. If you want I can also add `pointerenter`, but I don't have a pressing need for it.
- I moved the node version-related work to a different PR, see https://github.com/uber-web/mjolnir.js/pull/27.
- I tested the new events in the example app and they are working.
- I updated the docs that I could find, let me know if I missed other docs that should be updated.

**Comment by Pessimistress** [comment]
2.0.1 is published.

**Pull request merged**